File: frontend/gestionale-bertozzi/src/app/models/Anagrafiche/cliente.ts
import { PersonaleCliente } from './personale-cliente';

export class Cliente {
    id?: number;
    ragioneSociale?: string;
    codiceInterno?: string;
    partitaIva?: string;
    codiceFiscale?: string;
    indirizzo?: string;
    comune?: string;
    cap?: string;
    provincia?: string;
    nazione?: string;
    telefono?: string;
    email?: string;
    modalitaPagamentoId?: number;
    sdi?: string;
    tipo?: string;
    personale?: PersonaleCliente[];

    static map(cliente: any): Cliente {
        let c = Object.assign(new Cliente(), cliente) as Cliente;
        
        // Mappa l'array del personale se presente
        if (cliente.personale && Array.isArray(cliente.personale)) {
            c.personale = PersonaleCliente.mapArray(cliente.personale);
        }
        
        return c;
    }

    static mapArray(arr: any[]): Cliente[] {
        return (arr || []).map(c => Cliente.map(c));
    }
}

File: backend/NemesiLIB/Model/Anagrafiche/Cliente.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace NemesiLIB.Model.Anagrafiche
{
    public class Cliente
    {
        public int Id { get; set; }
        public string RagioneSociale { get; set; }
        public string CodiceInterno { get; set; }
        public string PartitaIva { get; set; }
        public string CodiceFiscale { get; set; }
        public string Indirizzo { get; set; }
        public string Comune { get; set; }
        public string CAP { get; set; }
        public string Provincia { get; set; }
        public string Nazione { get; set; }
        public string? Telefono { get; set; }
        public string? Email { get; set; }
        public int ModalitaPagamentoId { get; set; }
        public string? Sdi { get; set; }
        public string Tipo { get; set; }

        public virtual List<PersonaleCliente>? Personale { get; set; }
    }
}
